### PR TITLE
Truncate results from search_files to 500 chars max

### PR DESCRIPTION
**Pull request by mrubens**
This was crashing my extension when returning huge lines of text 😞 

----

> [!IMPORTANT]
> Truncate lines longer than 500 characters in search results to prevent crashes, with tests added for verification.
> 
>   - **Behavior**:
>     - Truncate lines longer than 500 characters in `regexSearchFiles` in `index.ts` to prevent crashes.
>     - Truncation adds " [truncated...]" to indicate truncation.
>   - **Functions**:
>     - Add `truncateLine` function in `index.ts` to handle line truncation with a default max length of 500 characters.
>     - Update `regexSearchFiles` to use `truncateLine` for both match and context lines.
>   - **Tests**:
>     - Add tests in `index.test.ts` to verify line truncation behavior, including custom max length and edge cases.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooVetGit%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for dfa019e7f443bf9997ed2ac2556f597a59e4bf77. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/services/ripgrep/__tests__/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// npx jest src/services/ripgrep/__tests__/index.test.ts
+
+import { describe, expect, it, jest } from "@jest/globals"
+
+// Mock implementation of the line truncation feature
+function truncateLine(line: string, maxLength: number): string {
```
[the remark this answers is not in the document]
Comment (mrubens) [reply src/services/ripgrep/__tests__/index.test.ts:6]: Ha oops, fixing

**Review by cte — APPROVED** [review]
